conf/modules.config

```diff
@@ -0,0 +1,179 @@
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Config file for defining DSL2 per module options and publishing paths
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Available keys to override module options:
+        ext.args   = Additional arguments appended to command in module.
+        ext.args2  = Second set of arguments appended to command in module (multi-tool modules).
+        ext.args3  = Third set of arguments appended to command in module (multi-tool modules).
+        ext.prefix = File name prefix for output files.
+----------------------------------------------------------------------------------------
+*/
+
+process {
+
+    publishDir = [
+        path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+    ]
+
+    withName: SAMPLESHEET_CHECK {
+        publishDir = [
+            path: { "${params.outdir}/pipeline_info" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
+    withName: CUSTOM_DUMPSOFTWAREVERSIONS {
+        publishDir = [
+            path: { "${params.outdir}/pipeline_info" },
+            mode: params.publish_dir_mode,
+            pattern: '*_versions.yml'
+        ]
+    }
+
+    withName: FASTQC {
+        ext.args = '--quiet'
+    }
+
+    withName: FASTP {
+        ext.prefix = { "${meta.id}.trimmed" }
+    }
+
+    withName: GFFREAD {
+        ext.args = "--keep-exon-attrs -F -T"
+    }
+
+    withName: BWA_MEM {
+        publishDir = [
+            [
+                path: { "${params.outdir}/${params.aligner}/log" },
+                mode: 'copy',
+                pattern: "out"
+            ],
+            [
+                path: { "${params.outdir}/${params.aligner}/log" },
+                mode: 'copy',
+                pattern: "tab"
+            ]
+        ]
+    }
+
+    withName: 'DRAGMAP_HASHTABLE' {
+        ext.when         = { !params.dragmap && params.aligner == "dragmap" }
```

Comment:
Hmm, I don't know. I think for things like

```
ext.when = { params.save_reference }
```

It's perfect for.

So in this case it doesn't make a lot of sense, but there is an if that handles this. I think some nf-core guidelines on this would be helpful, but I think fixing these are outside the scope of this release.

Made an issue to follow up on this #85 I think coming to some consensus on what makes it easiest to contribute to pipelines. There was also talk that relying on the `ext.when` could easily break in the future with the new config stuff.